src/librustc_typeck/diagnostics.rs

```diff
@@ -1457,6 +1457,42 @@ impl Foo for Bar {
 ```
 "##,
 
+E0327: r##"
+You cannot use a method instead of a constant expression as a pattern. Example
```

Comment:
Technically, "You cannot use associated items other than constant items as patterns. This includes method items". (methods aren't the only associated items)